### DISCUSSION TITLE
Graphics OR and NORMAL mode.

### DISCUSSION THREAD

**Discussion by agganitk** [board707/DMD_STM32#105] [General]
For P10 Monochrome display, I am using GRAPHICS_OR mode by commenting below lines in GRAPHICS_NORMAL.
**

**
This way I am able to display non english language characters properly, where some characters overlap eachother.
I want to have both FIXED and SCROLL screen at same time and able to do using SWAPBUFFERS

I have FIXED screen on left side. Scrolling text has non english language characters.
But by using Graphics_OR, the fixed screen portion starts showing both SCROLLING and FIXED text.
if I use Graphic_Normal, then the text in FIXED screen shows only fix text but then Scrolling text non english characters doesn't appear properly as they need OR function to display properly.
Can you suggest some solution? how to get this done.

**Comment by board707** [comment]
Hi
Could you provide some examples of the code? 
I don't quite understand what the problem is.
Pictures or videos of the screen will also be useful

**Comment by agganitk** [comment]


**Reply by board707** [reply]
> As the text reaches into A portion, the A portion starts displaying both Fixed text and scrolling text.
I mean A portion has both FIXED text and scrolling text LEDs glowing

Is it what you intended or not?
I still not understand why you need to edit GRAPHIC modes for it.

**Reply by agganitk** [reply]
Is it what you intended or not?
^^^^^^^^^^^^^^^^^^   I do not want that. I want A portion only has FIXED texts LED glowing.  Scrolling text should just become invisible once it enters A portion.

**Reply by board707** [reply]
> To display HINDI language...

It is very interesting. Last year I tried to display Devanagari with my library with some success. Do you have a custom Hindi font?

Returning to the issue - I think to help you I need a code example.
Could you provide a short (but complete ) sketch example to use a fixed and scrolling Hindi text on the matrix?

**Reply by agganitk** [reply]
Hi I am not using any custom font. It is all generated converting Devanagari ttf file to .h format.
Can you try with english character font.   you will see A portion showes led glowing for  both fixed and scrolling text and ultimately not readeable. 
 As I said we need to use Graphics OR for scrolling text but it messes up Fixed A portion.  I want to display only English characters in FIXED A portion, so no need for OR function.

**Comment by board707** [comment]
> you will see A portion showes led glowing for both fixed and scrolling text and ultimately not readeable.

I still think that you don't need to change GRAPHICS mode for it. It can be done with dual buffering mode.
See the example `examples/STM32F4_RP2040/dmd_rgb_dual_buf`. The code prints a fixed text "23" at the left part of display and scroll the text at the right.
For more complicated modes such as two strings, scrolling in opposite directions - see the discussion in issue #99
